### PR TITLE
5. Configurando las pruebas con Typescript - Resolved

### PR DESCRIPTION
**Pull request by JorgeVQA**
This is the resolution for 5. Configurando las pruebas con Typescript.   Please give me your thoughts on this part of the workshop.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -21,5 +22,25 @@
...
+            "cypress",
+            "node"
+        ]
+    },
+    "include": [
+        "cypress/**/*.ts"
+    , "src"],
+    "exclude": [
+        "node_modules/"
+    ]
```
Comment (holgiosalos) [inline package.json]: The code between the lines 30 to 45, is already in the "tsconfig.json" file, those are not needed in this file.

File: package.json [hunk 2]
```diff
@@ -4,7 +4,8 @@
   "description": "This is a Workshop about Cypress",
   "main": "index.js",
   "scripts": {
-    "test:open": "cypress open"
+    "test:open": "npx cypress open",
+    "test:run": "npx cypress run"
```
Comment (holgiosalos) [inline package.json]: the npx command is not needad as prefix for test:open or test:run scripts

File: tsconfig.json [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+            "node"
+        ]
+    },
+    "include": [
+        "cypress/**/*.ts"
+    ],
+    "exclude": [
+        "node_modules/"
+    ]
+}
```
Comment (holgiosalos) [inline tsconfig.json]: Add the EOF

File: package.json [hunk 4]
```diff
@@ -21,5 +22,25 @@
   "bugs": {
     "url": "https://github.com/JorgeVQA/cypress-training/issues"
   },
-  "homepage": "https://github.com/JorgeVQA/cypress-training#readme"
+  "homepage": "https://github.com/JorgeVQA/cypress-training#readme",
+  "devDependencies": {
+    "@types/jest": "^27.4.1",
```
Comment (holgiosalos) [inline package.json:27]: why did you need the @types/jest dependency?

File: package.json (hunk 1 above)
Comment (JorgeVQA) [reply package.json]: I will remove it, this is an exercise I was doing with Juan De La Hoz due the tsconfig file was not showing up as a .json.   Finally today worked so I will remove it here.   Thanks

File: package.json (hunk 2 above)
Comment (JorgeVQA) [reply package.json]: Checking progress with Juan De la Hoz, "cypress open" command didn't work though we used npx and it did.   I will check if woks as you suggested.   If not I will ask on Monday.

File: tsconfig.json (hunk 3 above)
Comment (JorgeVQA) [reply tsconfig.json]: Thanks, I will.

File: package.json (hunk 4 above)
Comment (JorgeVQA) [reply package.json:27]: When changed extension to test file from .js to .ts some reserved words were not recognized in the code, so I took suggestions where errors marked the code and there was a command I run and that was why that dependency was added.   I will check on Monday with you, Thanks.

File: package.json (hunk 1 above)
Comment (JorgeVQA) [reply package.json]: Fixed, I will send changes for you to check

File: package.json (hunk 2 above)
Comment (JorgeVQA) [reply package.json]: Fixed, I will send changes for you to check

Comment (JuanDelahozMiranda) [reply package.json]: Hi Guys, so, like we discuss, I suggest Jorge continue in JS, cause the TS maybe is missing dependencies.

File: package.json (hunk 4 above)
Comment (JuanDelahozMiranda) [reply package.json:27]: Like improve, you would set up these dependencies in your project, in this way you ensure that in other machine is set up properly

Improve_packageFile

-- try to run --
Not_Run_project_In_Other_Machine

'>>> 3 Doritos after >>>
Dependencies_Save

Run_Test

File: package.json (hunk 2 above)
Comment (JuanDelahozMiranda) [reply package.json]: Additionally, the command Cypress sometimes is not recognized like a global command, so is possible to use the 'npx' too in the command

**Review by JuanDelahozMiranda — CHANGES_REQUESTED** [review]
According to the conversation, need to apply some improves :)

File: package.json (hunk 4 above)
Comment (JorgeVQA) [reply package.json:27]: Fixed, I will update PR with a new commit.

**Review by JuanDelahozMiranda — APPROVED** [review]
